### PR TITLE
Added unit tests for lrucache

### PR DESCRIPTION
**Pull request by suja1956**
This PR contains unit tests for lrucache.go file. 
Below are the unit tests that I have added:
1. TestOverwriteValue: Tests that if a value is set for an existing key, the value is properly overwritten.
2. TestEviction: Verifies that when the cache exceeds its capacity, the least recently used item is evicted. This test uses a cache of size 2 to simulate the eviction.
3. TestMoveToFront: Ensures that when an item is accessed, it becomes the most recently used item. The least recently used item should be evicted after inserting a new one.
4. TestLRUBehavior: Tests more advanced LRU behavior by accessing different items and confirming that the least recently accessed item is the one that gets evicted when the cache is full.

Fixes: #10

### CODE REVIEW

**Comment by yp969803** [comment]
@suja1956 run 'make format' before commiting, also the tests are failing

**Comment by yp969803** [comment]
Also you can remove, TestGetSet function in  https://github.com/fordevio/wharf/blob/main/pkg/cache/lrucache_test.go

**Comment by yp969803** [comment]
also you can remove https://github.com/fordevio/wharf/blob/ad512cea5d684ab95c970ec8ab5ad9afc027ff31/pkg/cache/lrucache_test.go#L7

File: pkg/cache/lrucache_test.go [hunk 1]
```diff
@@ -22,3 +22,124 @@ func TestInvalidate(t *testing.T) {
...
+	if val != "value2" {
+		t.Errorf("Expected value2, got %s", val)
+	}
+
+	// Insert a new item, which should cause key1 (the least recently used) to be evicted
+	cache.Set("key3", "value3")
+
+	val = cache.Get("key1")
+	if val != nil {
+		t.Errorf("Expected nil (key1 evicted), got %s", val)
```
Comment (yp969803) [inline pkg/cache/lrucache_test.go:59]: Test failing here

Comment (suja1956) [reply pkg/cache/lrucache_test.go:59]: Will try to resolve.

**Comment by yp969803** [comment]
@suja1956 can you pull the current main branch, i have done some changes in lrucache functions

**Pull request closed without merging**